bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,536 @@
+                            } else if ((dpval[i].value) == 11) {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf("SWIRL"));
+                            } else if ((dpval[i].value) == 12) {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf("WIDE"));
+                            } else {
+                                updateState(CHANNEL_TYPE_VANESLR, StringType.valueOf(String.valueOf(dpval[i].value)));
+                            }
+                            break;
+                        case 9:
+                            int unit = Math.round((dpval[i].value) / 10);
+                            State stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                            updateState(CHANNEL_TYPE_TARGETTEMP, stateValue);
+                            break;
+                        case 10:
+                            unit = Math.round((dpval[i].value) / 10);
+                            stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                            updateState(CHANNEL_TYPE_AMBIENTTEMP, stateValue);
+                            break;
+                        case 37:
+                            unit = Math.round((dpval[i].value) / 10);
+                            stateValue = QuantityType.valueOf(unit, SIUnits.CELSIUS);
+                            updateState(CHANNEL_TYPE_OUTDOORTEMP, stateValue);
+                            break;
+                    }
+                }
+            }
+        }
+    }
+
+    public void login() {
+        String contentString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\"" + password
+                + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        if (response != null) {
+            Response resp = gson.fromJson(response, Response.class);
+            boolean success = resp.success;
+            if (success) {
+                Data data = gson.fromJson(resp.data.toString(), Data.class);
+                Id id = gson.fromJson(data.id.toString(), Id.class);
+                if (id.sessionID.toString() != null && !id.sessionID.toString().isEmpty()) {
+                    sessionId = id.sessionID.toString();
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            }
+        }
+    }
+
+    public @Nullable String logout() {
+        String contentString = "{\"command\":\"logout\",\"data\":{\"sessionID\":\"" + sessionId + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        return response;
+    }
+
+    public boolean getSuccess(String response) {
+        Response resp = gson.fromJson(response, Response.class);
+        boolean success = resp.success;
+        return success;
+    }
```

Comment:
I don't think you should have this method, parsing is too expensive to throw away the result on success. It would be better if you inlined this code in the callers and then got rid of this method.

Reply:
removed